test/Microsoft.ML.Tests/Scenarios/Api/Estimators/PredictAndMetadata.cs

```diff
@@ -37,22 +37,22 @@ void PredictAndMetadata()
 
             var testLoader = ml.Data.LoadFromTextFile(dataPath, TestDatasets.irisData.GetLoaderColumns(), separatorChar: ',', hasHeader: true);
             var testData = ml.Data.CreateEnumerable<IrisData>(testLoader, false);
-            
+
             // During prediction we will get Score column with 3 float values.
             // We need to find way to map each score to original label.
-            // In order to do what we need to get SlotNames from Score column.
-            // Slot names on top of Score column represent original labels for i-th value in Score array.
-            VBuffer<ReadOnlyMemory<char>> slotNames = default;
-            engine.OutputSchema[nameof(IrisPrediction.Score)].GetSlotNames(ref slotNames);
+            // In order to do what we need to get TrainingLabelValues from Score column.
+            // TrainingLabelValues on top of Score column represent original labels for i-th value in Score array.
+            VBuffer<ReadOnlyMemory<char>> originalLabels = default;
+            engine.OutputSchema[nameof(IrisPrediction.Score)].Annotations.GetValue(AnnotationUtils.Kinds.TrainingLabelValues, ref originalLabels);
```

Comment:
>TrainingLabelValues [](start = 105, length = 19)

Is the distinction with slot names that slots names *must* be text, while these might be any type?

Reply:
No we don't, but I can change that. Any reason while we want continue to propagate slotnames?

Reply:
Well, let's imagine I write out a text file, and I have this scores column. With slot names, I get a descriptive header. Without it I don't. Does that make sense?

Reply:
Only if original labels were string, but ok.
Just before I make crucial mistake.
Do you prefer to have two wrappers on top of multiclass scorer one for `TrainingLabelValues` one for `SlotNames` or you would prefer to extend `LabelNameBindableMapper` to support multiple getters/ metakinds?

Reply:
Multiple metadata kinds is good, thanks Ivan.

---
In reply to: [269742478](https://github.com/dotnet/machinelearning/pull/3101#discussion_r269742478) [](ancestors = 269742478)